### PR TITLE
AP_HAL_ChibiOS:add mazzy star drone hwdef

### PR DESCRIPTION
**Pull request by Huibean**
add mazzy star drone hwdef to master

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+# leave 2 sectors free
+FLASH_RESERVE_START_KB 96
+
+# board voltage
+STM32_VDD 330U
+
+# only one I2C bus
+I2C_ORDER I2C1
+
+# order of UARTs (and USB), USART3 should be in second place to map order with the board's silk screen
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat:32]: comment on USART3 is wrong

File: libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+SERIAL_ORDER OTG1 UART4 USART1 USART2 USART3 UART7 USART6
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat:33]: UART4 is listed as GPS2 below, but is setup as SERIAL1 here

File: libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef-bl.dat [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+FLASH_RESERVE_START_KB 0
+
+# the location where the bootloader will put the firmware
+FLASH_BOOTLOADER_LOAD_KB 96
+
+# board voltage
+STM32_VDD 330U
+
+# order of UARTs (and USB)
+UART_ORDER OTG1
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef-bl.dat:34]: needs to be SERIAL_ORDER

File: libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat (hunk 2 above)
Comment (Huibean) [reply libraries/AP_HAL_ChibiOS/hwdef/MazzyStarDrone/hwdef.dat]: I have updated the SERIAL_ORDER, USART3 and USART4 is reverse in the board

**Review by tridge — APPROVED** [review]

**Pull request merged**